src/randgraphs.jl

```diff
@@ -244,7 +244,64 @@ function random_regular_digraph(n::Int, k::Int; dir::Symbol=:out, seed::Int=-1)
     end
 end
 
-"""StochasticBlockModel(n,nodemap,affinities)
+doc"""
+    stochastic_block_model(c::Matrix{Float64}, n::Vector{Int}; seed::Int = -1)
+    stochastic_block_model(cin::Float64, coff::Float64, n::Vector{Int}; seed::Int = -1)
+
+Returns a Graph generated according to the Stochastic Block Model (SBM).
+
+`c[a,b]` : Mean number of neighbors of a vertex in block `a` belonging to block `b`.
+           Only the upper triangular part is considered, since the lower traingular is
+           determined by c[b,a] = c[a,b] * n[a]/n[b].
+`n[a]` : Number of vertices in block `a`
+
+The second form samples from a SBM with `c[a,a]=cin`, and `c[a,b]=coff`.
+
+For a dynamic version of the SBM see the StochasticBlockModel type and
+related functions.
+"""
+function stochastic_block_model{T}(c::Matrix{T}, n::Vector{Int}; seed::Int = -1)
+    @assert size(c,1) == length(n)
+    @assert size(c,2) == length(n)
+    # init dsfmt generator without altering GLOBAL_RNG
+    seed > 0 && Base.dSFMT.dsfmt_gv_init_by_array(MersenneTwister(seed).seed+1)
+    rng =  seed > 0 ? MersenneTwister(seed) : MersenneTwister()
+
+    N = sum(n)
+    K = length(n)
+    nedg = zeros(Int,K, K)
+    g = Graph(N)
+    cum = [sum(n[1:a]) for a=0:K]
+    for a=1:K
+        ra = cum[a]+1:cum[a+1]
+        for b=a:K
+            @assert a==b? c[a,b] <= n[b]-1 : c[a,b] <= n[b]
```

Comment:
This would probably be better with a real error message: why is this condition incorrect, and how do we avoid it?